comp/forwarder/defaultforwarder/resolver/domain_resolver.go

```diff
@@ -136,15 +150,20 @@ func updateAdditionalEndpoints(resolver DomainResolver, setting string, config c
 	removed := missing(oldKeys, newKeys)
 	added := missing(newKeys, oldKeys)
 
+	health := resolver.GetForwarderHealth()
+	if health != nil {
+		health.UpdateAPIKeys(resolver.GetBaseDomain(), removed, added)
+	}
+
```

Comment:
While this is correct, manually calling `scrubKeys` within following log lines feels a bit brittle, if someone were to add or a change log line without paying too much attention it isn't ideal.

What do you think of doing the scrubbing before the logging logic, here?

```suggestion

removed = scrubKeys(removed)
added = scrubKeys(added)
```